### ISSUE TITLE
Engine doesn't make a move

### ISSUE THREAD

**Issue by Aspiration-to-Stars** [careless25/lichess-bot#19]
I downloaded the new code, now it accepts challenges, but doesn't make moves. 
https://pastebin.com/SW6RHiDN

**Comment by HarshithBolar** [comment]
Worth mentioning that it only happens when playing as black

**Comment by Aspiration-to-Stars** [comment]
For me this happens in both cases: with white and black pieces.

**Comment by ZirconiumX** [comment]
@Aspiration-to-Stars That sounds like you have a raw number in your config.yml instead of a string. In the engine section you need to put quotes around all numbers.

**Comment by Aspiration-to-Stars** [comment]
@ZirconiumX, yes, I did already. This is what's happening when I do that:
https://pastebin.com/ftQDGctC

**Comment by ZirconiumX** [comment]
That's a bug in your engine. Have you checked it runs normally?

**Comment by Aspiration-to-Stars** [comment]
But that's leelaZero from the official site... Is there a way to check it without installing some special programs?

**Comment by ana-ka** [comment]
Happens for me too sometimes. Maybe this is related:

`Welcome ana-ka-bot!
Something went wrong. Game is starting and we don't have a queued process
`

**Comment by Aspiration-to-Stars** [comment]
How to install python3 to PATH? I have python installed to PATH and it's last version from the official site, but this (virtualenv .venv -p python3) fails... Trying last code.

**Comment by careless25** [comment]
@Aspiration-to-Stars Yes you can just write python in that case.

**Comment by careless25** [comment]
Guys, the errors you are getting would be helpful. Copy paste them here.

**Comment by Aspiration-to-Stars** [comment]
@careless25, this is what's happening after I wrote "python":
PS C:\Users\Имя пользователя> cd E:\chess\lichess-bot
PS E:\chess\lichess-bot> virtualenv .venv -p python
Running virtualenv with interpreter C:\Users\Имя пользователя\AppData\Local\Programs\Python\Python36-32\python.exe
Using base prefix 'C:\\Users\\Имя пользователя\\AppData\\Local\\Programs\\Python\\Python36-32'
New python executable in E:\chess\lichess-bot\.venv\Scripts\python.exe
Failed to import the site module
Traceback (most recent call last):
  File "E:\chess\lichess-bot\.venv\lib\site.py", line 703, in <module>
    main()
  File "E:\chess\lichess-bot\.venv\lib\site.py", line 670, in main
    virtual_install_main_packages()
  File "E:\chess\lichess-bot\.venv\lib\site.py", line 554, in virtual_install_main_packages
    sys.real_prefix = f.read().strip()
  File "E:\chess\lichess-bot\.venv\lib\encodings\cp1251.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x98 in position 10: character maps to <undefined>
ERROR: The executable E:\chess\lichess-bot\.venv\Scripts\python.exe is not functioning
ERROR: It thinks sys.prefix is 'e:\\chess\\lichess-bot' (should be 'e:\\chess\\lichess-bot\\.venv')
ERROR: virtualenv is not compatible with this system or executable
Note: some Windows users have reported this error when they installed Python for "Only this user" or have multiple versi
ons of Python installed. Copying the appropriate PythonXX.dll to the virtualenv Scripts/ directory may fix this problem.

PS E:\chess\lichess-bot>

**Comment by careless25** [comment]
@Aspiration-to-Stars `virtualenv .\.venv -p python`
Try that. ugh windows

**Comment by careless25** [comment]
You missed a slash in your command

**Comment by Aspiration-to-Stars** [comment]
@careless25, somehow it has been copied without slash, but it had slash.

**Comment by ana-ka** [comment]
@careless25 there are no exceptions or anything. Just the logger output.

> Welcome ana-ka-bot!
Something went wrong. Game is starting and we don't have a queued process
--- Process Used. Total Queued: 0. Total Used: 1
+++ https://lichess.org/2DFsUmUU/white Bullet vs AI level 7
--- https://lichess.org/2DFsUmUU/white Game over
+++ Process Free. Total Queued: 0. Total Used: 0
Something went wrong. Game is starting and we don't have a queued process
--- Process Used. Total Queued: 0. Total Used: 1
+++ https://lichess.org/m76zJwFc/black Bullet vs AI level 7
--- https://lichess.org/m76zJwFc/black Game over
+++ Process Free. Total Queued: 0. Total Used: 0
Something went wrong. Game is starting and we don't have a queued process
--- Process Used. Total Queued: 0. Total Used: 1
+++ https://lichess.org/853KHImf/black Bullet vs AI level 8
--- https://lichess.org/853KHImf/black Game over
+++ Process Free. Total Queued: 0. Total Used: 0

However, the engine made a move in some of these games. It appears to work if it worked once, so to speak. First game: no move. Abort and revenge: appears to work. Can't make out a clear pattern yet.

**Comment by careless25** [comment]
Which OS and which engine? Can you log more output to figure out where it gets stuck exactly?

**Comment by Aspiration-to-Stars** [comment]
I have Windows 7 64-bit. Used leela-gpu 0.7 release. Now, after some help I was able to run it for usual engines (though that's maybe older version of repository). But when I try to run leelaZero now, it says:
"libopenblas.dll is missing on this computer". 
And this is written:
Welcome Ingot!
Traceback (most recent call last):
  File "E:\chess\leela-bot\logging_pool.py", line 16, in __call__
    result = self.__callable(*args, **kwargs)
  File "E:\chess\leela-bot\main.py", line 83, in play_game
    engine, info_handler = setup_engine(engine_path, board, weights, threads)
  File "E:\chess\leela-bot\main.py", line 155, in setup_engine
    engine.uci()
  File "C:\Users\Имя пользователя\AppData\Local\Programs\Python\Python36-32\lib\
site-packages\chess\uci.py", line 688, in uci
    return self._queue_command(command, async_callback)
  File "C:\Users\Имя пользователя\AppData\Local\Programs\Python\Python36-32\lib\
site-packages\chess\uci.py", line 666, in _queue_command
    return future.result(timeout=FUTURE_POLL_TIMEOUT)
  File "C:\Users\Имя пользователя\AppData\Local\Programs\Python\Python36-32\lib\
concurrent\futures\_base.py", line 432, in result
    return self.__get_result()
  File "C:\Users\Имя пользователя\AppData\Local\Programs\Python\Python36-32\lib\
concurrent\futures\_base.py", line 384, in __get_result
    raise self._exception
  File "C:\Users\Имя пользователя\AppData\Local\Programs\Python\Python36-32\lib\
concurrent\futures\thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "C:\Users\Имя пользователя\AppData\Local\Programs\Python\Python36-32\lib\
site-packages\chess\uci.py", line 686, in command
    raise EngineTerminatedException()
chess.engine.EngineTerminatedException

    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.
    No challenge in the queue.

**Comment by Aspiration-to-Stars** [comment]
Ok, it works now due to all amount of help I received :)
Now I want to ask, how many threads can I set, if I use GPU version? And what is stronger: GPU or CPU version?

**Comment by ddugovic** [comment]
I've heard that Leela-GPU is about 10x faster than Leela-CPU.

**Comment by Aspiration-to-Stars** [comment]
That's nice to hear!

**Comment by Aspiration-to-Stars** [comment]
Now I have the problem that it doesn't move when playing as black, if I challenge stockfish. It works, though, when I challenge a player. That's strange...

**Comment by careless25** [comment]
This should be fixed now. Pull master again. 
Waiting on confirmation before closing this issue.

**Comment by Aspiration-to-Stars** [comment]
Is it possible and necessary to do these commands after requirements were already installed?
pip install virtualenv
virtualenv .venv -p python
./.venv/Scripts/activate
Now it's running with an older version without virtualenv...
And I'm afraid it all will crush now, if I install something else...

**Comment by Aspiration-to-Stars** [comment]
Now "virtualenv .venv -p python" worked after I reinstalled python for all users, and installed it in path without spaces. But now I'm stuck at this: "./.venv/Scripts/activate". It says can't load the file  E:\chess\lichess-bot-master\.venv\Scripts\activate.ps1 because executing scripts is forbidden for this system.

**Comment by Aspiration-to-Stars** [comment]
Ok, seems I've done it after allowing scripts on this computer with "Set-ExecutionPolicy RemoteSigned".

**Comment by Aspiration-to-Stars** [comment]
Now this happens:

(.venv) PS E:\chess\lichess-bot-master> python main.py
Traceback (most recent call last):
  File "main.py", line 185, in <module>
    CONFIG = load_config()
  File "E:\chess\lichess-bot-master\config.py", line 42, in load_config
    raise Exception("The engine specified does not exist.")
Exception: The engine specified does not exist.
(.venv) PS E:\chess\lichess-bot-master>

**Comment by Aspiration-to-Stars** [comment]
Ok, after adding "exe" to engine's name in yml file it works.
Now the only problem for me is that it's playing slow at the end, but I've made another issue on that.

**Comment by careless25** [comment]
So this issue is solved @Aspiration-to-Stars ?

**Comment by Aspiration-to-Stars** [comment]
@careless25, yes, it's solved for me. Thank you!
